src/overlays/actors/ovl_Obj_Tokeidai/z_obj_tokeidai.c

```diff
@@ -50,87 +42,803 @@ const ActorInit Obj_Tokeidai_InitVars = {
+
+    Actor_ProcessInitChain(&this->actor, D_80AB49E0);
+    this->actionFunc = ObjTokeidai_DoNothing;
+    this->opaDList = NULL;
+    this->xluDList = NULL;
+    this->xRotation = 0;
+    this->yTranslation = 0;
+    this->clockFaceZTranslation = 0;
+    this->currentTime = gSaveContext.time;
+    this->actor.home.rot.x = 0;
+
+    switch (OBJ_TOKEIDAI_TYPE(&this->actor)) {
+        case OBJ_TOKEIDAI_TYPE_TOWER_GEAR_TERMINA_FIELD:
+            Actor_SetScale(&this->actor, 0.15f);
+            ObjTokeidai_TowerGear_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_TOWER_GEAR_CLOCK_TOWN:
+            ObjTokeidai_TowerGear_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_UNUSED_WALL:
+            this->opaDList = D_0600D388;
+            break;
+        case OBJ_TOKEIDAI_TYPE_TOWER_WALLS_TERMINA_FIELD:
+            Actor_SetScale(&this->actor, 1.0f);
+            this->opaDList = D_06009A08;
+            this->actionFunc = ObjTokeidai_Walls_Idle;
+            break;
+        case OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_TERMINA_FIELD:
+            Actor_SetScale(&this->actor, 0.15f);
+            ObjTokeidai_TowerClock_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_TOWER_CLOCK_CLOCK_TOWN:
+            ObjTokeidai_TowerClock_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_WALL_CLOCK:
+            Actor_SetScale(&this->actor, 0.02f);
+            this->actor.draw = ObjTokeidai_Clock_Draw;
+            ObjTokeidai_Clock_Init(this);
+            this->actionFunc = ObjTokeidai_WallClock_Idle;
+            break;
+        case OBJ_TOKEIDAI_TYPE_SMALL_WALL_CLOCK:
+            Actor_SetScale(&this->actor, 0.01f);
+            this->actor.draw = ObjTokeidai_Clock_Draw;
+            ObjTokeidai_Clock_Init(this);
+            this->actionFunc = ObjTokeidai_WallClock_Idle;
+            break;
+        case OBJ_TOKEIDAI_TYPE_COUNTERWEIGHT_TERMINA_FIELD:
+            Actor_SetScale(&this->actor, 0.15f);
+            ObjTokeidai_Counterweight_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_COUNTERWEIGHT_CLOCK_TOWN:
+            ObjTokeidai_Counterweight_Init(this, globalCtx);
+            break;
+        case OBJ_TOKEIDAI_TYPE_STAIRCASE_INTO_TOWER:
+            this->opaDList = D_0600D8E8;
+            this->xluDList = D_0600D8E0;
+            this->actionFunc = ObjTokeidai_StaircaseIntoTower_Idle;
+            break;
+    }
+}
+
+void ObjTokeidai_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void ObjTokeidai_RotateOnMinuteChange(ObjTokeidai* this, s32 playSound) {
+    s32 currentMinute;
+
+    currentMinute = OBJ_TOKEIDAI_GET_CURRENT_MINUTE(this);
+    if (currentMinute != this->clockMinute) {
+        if (this->outerRingOrGearRotationTimer == 8 && playSound) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CLOCK_TOWER_SECOND_HAND);
+        }
+
+        if (this->outerRingOrGearRotationTimer >= 9) {
```

Comment:
```suggestion
        if (this->outerRingOrGearRotationTimer > 8) {
```
I think this would look better since 8 is used above